chromium_src/base/feature_list.cc

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/feature_list.h"
+#include "base/feature_override.h"
+#include "base/no_destructor.h"
+
+namespace base {
+namespace {
+
+using FeatureDefaultStateOverrides =
+    base::flat_map<const Feature*, FeatureState>;
+
+FeatureDefaultStateOverrides& GetFeatureDefaultStateOverrides() {
+  static base::NoDestructor<FeatureDefaultStateOverrides> overrides;
+  return *overrides;
+}
+
+FeatureState GetDefaultOrOverriddenState(const Feature& feature) {
+  const auto& overrides = GetFeatureDefaultStateOverrides();
+  auto override_it = overrides.find(&feature);
+  return override_it == overrides.end() ? feature.default_state
+                                        : override_it->second;
+}
+
+}  // namespace
+
+namespace internal {
+
+FeatureDefaultStateOverride::FeatureDefaultStateOverride(
```

Comment:
I am probably missing something obvious here, but why do this via a class and construct all those global objects instead of using a global function to populate the map?

Reply:
The question is at what point you have to populate the map and how to make sure it will happen in all processes. The global construction approach make it so in each process (utility, renderer, etc), the map is populated at the very start regardless of what Chromium/Brave delegate is called or not.

Reply:
Oh I see, yep, that totally makes sense. 👍